app/models/github_status.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+class GithubStatus
```

Comment:
we already have commit_status ... this feels like the same code all over again ... can we unify ?

Reply:
That would be a larger refactoring, and I mostly want to fix this UX problem right now...

Reply:
It seems that `CommitStatus` is deeply linked to stages...

Reply:
can commit-status use this then ?

Reply:
I can give it a try.